packages/fetchai/skills/erc1155_client/behaviours.py

```diff
@@ -0,0 +1,127 @@
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains the behaviour for the erc-1155 client skill."""
+
+from typing import cast
+
+from aea.crypto.ethereum import ETHEREUM
+from aea.crypto.fetchai import FETCHAI
+from aea.skills.behaviours import TickerBehaviour
+
+from packages.fetchai.protocols.oef.message import OEFMessage
+from packages.fetchai.protocols.oef.serialization import DEFAULT_OEF, OEFSerializer
+from packages.fetchai.skills.erc1155_client.strategy import Strategy
+
+DEFAULT_SEARCH_INTERVAL = 5.0
+
+
+class MySearchBehaviour(TickerBehaviour):
+    """This class implements a search behaviour."""
+
+    def __init__(self, **kwargs):
+        """Initialize the search behaviour."""
+        search_interval = cast(
+            float, kwargs.pop("search_interval", DEFAULT_SEARCH_INTERVAL)
+        )
+        super().__init__(tick_interval=search_interval, **kwargs)
+
+    def setup(self) -> None:
+        """Implement the setup for the behaviour."""
+        if self.context.ledger_apis.has_fetchai:
+            fet_balance = self.context.ledger_apis.token_balance(
+                FETCHAI, cast(str, self.context.agent_addresses.get(FETCHAI))
+            )
+            if fet_balance > 0:
+                self.context.logger.info(
+                    "[{}]: starting balance on fetchai ledger={}.".format(
+                        self.context.agent_name, fet_balance
+                    )
+                )
+            else:
+                self.context.logger.warning(
+                    "[{}]: you have no starting balance on fetchai ledger!".format(
+                        self.context.agent_name
+                    )
+                )
+                # TODO: deregister skill from filter
+
+        if self.context.ledger_apis.has_ethereum:
+            eth_balance = self.context.ledger_apis.token_balance(
+                ETHEREUM, cast(str, self.context.agent_addresses.get(ETHEREUM))
+            )
+            if eth_balance > 0:
+                self.context.logger.info(
+                    "[{}]: starting balance on ethereum ledger={}.".format(
+                        self.context.agent_name, eth_balance
+                    )
+                )
+            else:
+                self.context.logger.warning(
+                    "[{}]: you have no starting balance on ethereum ledger!".format(
+                        self.context.agent_name
+                    )
+                )
+                # TODO: deregister skill from filter
```

Comment:
This deregistering is now possible: https://github.com/fetchai/agents-aea/blob/develop/aea/skills/base.py#L99...just set it to `False`